arch/cpu/nrf/net/nrf-ieee-driver-arch.c

```diff
@@ -0,0 +1,1065 @@
+  conf.big_endian = false;
+  conf.whiteen = false;
+  conf.maxlen = MPDU_LEN;
+
+  nrf_radio_packet_configure(NRF_RADIO, &conf);
+}
+/*---------------------------------------------------------------------------*/
+static void
+setup_interrupts(void)
+{
+  int_master_status_t stat;
+  nrf_radio_int_mask_t interrupts = 0;
+
+  stat = critical_enter();
+
+  if(!rf_config.poll_mode) {
+    nrf_radio_event_clear(NRF_RADIO, NRF_RADIO_EVENT_CRCOK);
+    nrf_radio_event_clear(NRF_RADIO, NRF_RADIO_EVENT_CRCERROR);
+    interrupts |= NRF_RADIO_INT_CRCOK_MASK | NRF_RADIO_INT_CRCERROR_MASK;
+  }
+
+  /* Make sure all interrupts are disabled before we enable selectively */
+  nrf_radio_int_disable(NRF_RADIO, 0xFFFFFFFF);
+  NVIC_ClearPendingIRQ(RADIO_IRQn);
+
+  if(interrupts) {
+    nrf_radio_int_enable(NRF_RADIO, interrupts);
+    NVIC_EnableIRQ(RADIO_IRQn);
+  } else {
+    /* No radio interrupts required. Make sure they are all off at the NVIC */
+    NVIC_DisableIRQ(RADIO_IRQn);
+  }
+
+  critical_exit(stat);
+}
+/*---------------------------------------------------------------------------*/
+/*
+ * Set up timestamping with PPI:
+ * - Programme Channel 1 for RADIO->FRAMESTART--->TIMER0->CAPTURE[3]
+ * - Programme Channel 2 for RADIO->END--->TIMER0->CAPTURE[2]
+ */
+static void
+setup_ppi_timestamping(void)
+{
+  nrfx_gppi_channel_endpoints_setup(
+    NRF_PPI_FRAMESTART_CHANNEL,
+    nrf_radio_event_address_get(NRF_RADIO, NRF_RADIO_EVENT_FRAMESTART),
+    nrf_timer_task_address_get(NRF_TIMER0, NRF_TIMER_TASK_CAPTURE3));
+  nrfx_gppi_channel_endpoints_setup(
+    NRF_PPI_END_CHANNEL,
+    nrf_radio_event_address_get(NRF_RADIO, NRF_RADIO_EVENT_END),
+    nrf_timer_task_address_get(NRF_TIMER0, NRF_TIMER_TASK_CAPTURE2));
+  nrfx_gppi_channels_enable(1uL << NRF_PPI_FRAMESTART_CHANNEL
+                            | 1uL << NRF_PPI_END_CHANNEL);
+}
+/*---------------------------------------------------------------------------*/
+static void
+set_poll_mode(bool enable)
+{
+  rf_config.poll_mode = enable;
+  setup_interrupts();
+}
+/*---------------------------------------------------------------------------*/
+static void
+rx_buf_clear(void)
+{
+  memset(&rx_buf, 0, sizeof(rx_buf));
+}
+/*---------------------------------------------------------------------------*/
+static void
+rx_events_clear()
+{
+  nrf_radio_event_clear(NRF_RADIO, NRF_RADIO_EVENT_FRAMESTART);
+  nrf_radio_event_clear(NRF_RADIO, NRF_RADIO_EVENT_END);
+  nrf_radio_event_clear(NRF_RADIO, NRF_RADIO_EVENT_PHYEND);
+  nrf_radio_event_clear(NRF_RADIO, NRF_RADIO_EVENT_CRCERROR);
+  nrf_radio_event_clear(NRF_RADIO, NRF_RADIO_EVENT_CRCOK);
+}
+/*---------------------------------------------------------------------------*/
+/*
+ * Powering off the peripheral will reset all registers to default values
+ * This function here must be called at every power on to set the radio in a
+ * known state
+ */
+static void
+configure(void)
+{
+  nrf_radio_mode_set(NRF_RADIO, NRF_RADIO_MODE_IEEE802154_250KBIT);
```

Comment:
You might want to handle [nRF5340 anomaly 117](https://infocenter.nordicsemi.com/index.jsp?topic=%2Ferrata_nRF5340_Rev1%2FERR%2FnRF5340%2FRev1%2Flatest%2Fanomaly_340_117.html&anchor=anomaly_340_117) here.

Reply:
Agree, will do it.